main.go

```diff
@@ -64,22 +64,40 @@ func main() {
 	go serveMetrics(cfg.MetricsAddress)
 	go handleSigterm(stopChan)
 
-	client, err := newClient(cfg)
-	if err != nil {
-		log.Fatal(err)
+	var fakeSrcOnly = false
+	for _, svcType := range cfg.Sources {
+		switch svcType {
+		case "fake":
+			fqdn := cfg.FqdnTemplate
+			fakeSource, err := source.NewFakeSource(fqdn)
+			if err != nil {
+				log.Fatal(err)
+			}
+			source.Register("fake", fakeSource)
+
+			fakeSrcOnly = true
+			log.Warn("Fake source registered; will not register any other sources.")
+		}
 	}
 
-	serviceSource, err := source.NewServiceSource(client, cfg.Namespace, cfg.FqdnTemplate, cfg.Compatibility)
-	if err != nil {
-		log.Fatal(err)
-	}
-	source.Register("service", serviceSource)
+	if !fakeSrcOnly {
+		client, err := newClient(cfg)
```

Comment:
Sure!  I'm new-ish to golang; had some problems initially trying to check `if client == nil`, but I think the correct way to do that is `if client == kubernetes.Clientset{}`?

Then, before creation of `service` and `ingress` sources, we could check if client already exists, and instantiate if not.

Plus side: this makes client creation implicit.  Minus side: it makes client creation implicit.

I'll take a shot at that, see what you think.